linode/manager
# test: [M3-8435] - Cypress integration test for Object Storage Gen2: E3 Endpoint

## Changes  🔄
- New Integration Test for Object Storage Gen2: E3 Endpoint - Bucket Creation

## Target release date 🗓️
N/A

## How to test 🧪
- `yarn cy:run -s "cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts"`

Requirements:
- Visiting /object-storage/buckets/create causes "Create Bucket" drawer to open
- Users can select "E3" endpoint type for regions that support E3 endpoint types
    -  This will require mocking the response to the API endpoint types request to ensure that "E3" is available for the selected region
- Upon selecting "E3" endpoint type, rate limit table appears
    -  Rate limit table contains expected data
        -  (This data is currently pending, but notably it should differ from E2's rate limiting info)
- Informational notice text is displayed alongside rate limit table:
    -  Specifies the maximum Requests Per Second (RPS) for a bucket. To increase it to High, open a support ticket. Understand bucket rate limits.
- Upon clicking "Create Bucket" with "E3" endpoint type selected, Cloud fires an API bucket create request with the following data in its payload:
    -  cors_enabled is false
    -  endpoint_type is 'E3'
    -  s3_endpoint is present if corresponding endpoint mock contains non-null s3_endpoint
        -  At least one of these subtasks should test the case when s3_endpoint is specified, and at least one of these subtasks should test the case when s3_endpoint is omitted
-  Upon creating bucket, the "Create Bucket" drawer closes and new bucket appears in the landing page
    -  The correct information is displayed alongside the new bucket:
        -  Label
        -  Region
        -  Endpoint type
        -  Endpoint URL

## As an Author I have considered 🤔
*Check all that apply*
